base/libgit2/index.jl

```diff
@@ -112,9 +112,12 @@ function Base.count(idx::GitIndex)
 end
 
 function Base.getindex(idx::GitIndex, i::Csize_t)
+    if i < 1 || i >= count(idx)
+        throw(BoundsError())
```

Comment:
Same question as above, and should pass the `BoundsError` the object and the index. Same for the error in `Base.getindex(status::GitStatus, i::Csize_t)`.